### ISSUE TITLE
qt5 (5.5.0) failed to build on OSX 10.10.4

### ISSUE THREAD

**Issue by bagong** [Homebrew/homebrew#42217]
The build ends prematurely, I get this output on the command line:

==> ./configure -prefix /usr/local/Cellar/qt5/5.5.0 -system-zlib -securetransport -qt-libpng -qt-libjpeg -no-rpath -no-openssl -confirm-license -opensour
==> make
==> make install
==> make docs
make[4]: *** [sub-corelib-prepare_docs] Error 2
make[3]: *** [sub-src-prepare_docs] Error 2
make[2]: *** [module-qtbase-prepare_docs] Error 2
make[1]: *** [html_docs] Error 2
make: *** [docs] Error 2

I created a gist with brew gist-logs qt5 here:

https://gist.github.com/630fce5cdcef022e1b42

**Comment by blebo** [comment]
I get a failure at the same point. Previous brew install of Qt5 (v5.4.0) just days before the 5.5.0 update worked without any issues.

**Comment by DomT4** [comment]
It would seem there's an issue with the optional documentation. You can work around it by not invoking the documentation option for now. The error is rather strange:

```
dyld: Library not loaded: QtXml.framework/Versions/5/QtXml
  Referenced from: /usr/local/Cellar/qt5/5.5.0/bin/qdoc
  Reason: image not found
```

After installation executing `qdoc` works as expected, and it has the correct path. I don't build with documentation personally, so I'm not sure what has changed between 5.4.0 & 5.5.0 in regards to this step. It's possibly worth checking Qt's bug tracker for any similar issues.

**Comment by beemaster** [comment]
Could someone explain why all these arguments are needed? I guess this is legacy, and should probably be removed?
"-system-zlib -securetransport -qt-libpng -qt-libjpeg -no-rpath -no-openssl"

**Comment by mikemcquaid** [comment]
* `-system-zlib` - build against the system `zlib` to save compilation time. Maybe unnecessary but nice to use OS X native libraries where they cause no problems.
* `-securetransport` - Use Apple's SecureTransport instead of the (very) outdated system OpenSSL which has security issues.
* `-qt-libpng` - use Qt's `libpng` instead of a random one that can be found from e.g. X11, MacPorts or Homebrew
* `-qt-libjpeg` - use Qt's `libjpeg` instead of a random one that can be found from e.g. X11, MacPorts or Homebrew
* `-no-rpath` - fixes some `rpath` issues we saw with Qt 5.5.0 and was needed to build
* `-no-openssl` - Don't use the (very) outdated system OpenSSL which has security issues.

**Comment by beemaster** [comment]
Thank you for detailed explanation. Maybe rpath issues were solved here? https://bugreports.qt.io/browse/QTBUG-46660
By using -no-rpath it feels like rpath is disabled completely which is not a good thing

**Comment by mikemcquaid** [comment]
> Thank you for detailed explanation. Maybe rpath issues were solved here? https://bugreports.qt.io/browse/QTBUG-46660
> By using -no-rpath it feels like rpath is disabled completely which is not a good thing

@beemaster If you can test locally and find there's no more need to use `-no-rpath` we'd welcome a PR, thanks!

**Comment by DomT4** [comment]
Some additional notes:

* Qt5 will be adopting `-secure-transport -no-openssl` as default as of Qt 5.6's release (#41359).

* Qt's response towards changing Qt5 so we could build against a modern OpenSSL with support for ciphers above the TLS1.0 level and regularly security updates was largely "Eh". TLS1.0 is as secure as the 0.9.8 branch of OpenSSL is going to get. Secure Transport is an even better solution than ability to set custom OpenSSL.

* `-no-rpath` - See [this comment](https://github.com/Homebrew/homebrew/pull/41252#issuecomment-118557153). At the time of merge, it was breaking [these two formula](https://github.com/Homebrew/homebrew/pull/41252#issuecomment-118516233) *and* seemed to be causing a problem where we couldn't untar the Qt5 bottle.

**Comment by mkozjak** [comment]
Any news about --with-docs instalation problem on OS X 10.10.5 and qt5 5.5.0?
```
make[4]: *** [sub-corelib-prepare_docs] Error 2
make[3]: *** [sub-src-prepare_docs] Error 2
make[2]: *** [module-qtbase-prepare_docs] Error 2
make[1]: *** [html_docs] Error 2
make: *** [docs] Error 2
```

**Comment by DomT4** [comment]
No. Someone needs to kick this to Qt, really. The 5.5.0 release has quite a few problems that we're tracking, and hoping are resolved in 5.5.1. Docs slipped through the net as a failure point because it is optional and we don't test non-default options.

**Comment by UniqMartin** [comment]
This should have been resolved in the latest update to the `qt5` formula. Please give it another try with `brew update` followed by `brew install qt5 --with-docs`.
